src/driver-asus.c

```diff
@@ -189,51 +192,51 @@ static int
 asus_driver_save_profile(struct ratbag_device *device, struct ratbag_profile *profile)
 {
 	int rc = 0;
-	const int8_t *bmi = ratbag_device_data_asus_get_button_mapping(device->data);
-	struct asus_button *asus_button;
 	struct ratbag_button *button;
 	struct ratbag_led *led;
 	struct ratbag_resolution *resolution;
+	struct asus_data *drv_data = ratbag_get_drv_data(device);
 
 	/* set buttons */
 	ratbag_profile_for_each_button(profile, button) {
 		if (!button->dirty)
 			continue;
 
-		int asus_index = bmi[button->index] != -1 ? bmi[button->index] : ASUS_BUTTON_MAPPING_INDEX[button->index];
-		int asus_code;
+		int asus_index = drv_data->button_indices[button->index];
 		if (asus_index == -1) {
 			log_debug(device->ratbag, "No mapping for button %d\n", button->index);
 			continue;
 		}
 
-		log_debug(device->ratbag, "Button %d (%d) changed\n", button->index, asus_index);
+		int asus_code_src = drv_data->button_mapping[asus_index];
```

Comment:
This is missing handling of the sentinel `-1` value.
Also, it would be nice to make this a u8 from start, so perhaps something like this would be good:
```suggestion
		uint8_t asus_code_src = 0;
		{
			rc = drv_data->button_mapping[asus_index];
			if (rc == -1) {
				log_debug(device->ratbag, "No mapping for button %d\n", button->index);
				continue;
			}
			asus_code_src = (uint8_t)rc;
		}

```